src/gt4py/ir/utils.py

```diff
@@ -69,10 +69,7 @@ def make_field_ref(name: str, offset=(0, 0, 0), *, axes_names=None):
     return FieldRef(name=name, offset=offset)
 
 
-def make_axis_interval(bounds: tuple, *, offset_limit: int = 2):
-    assert isinstance(bounds[0], (VarRef, UnaryOpExpr, BinOpExpr)) or (
-        isinstance(bounds[0], int) and abs(bounds[0]) <= offset_limit
-    )
+def make_axis_interval(bounds: tuple):
```

Comment:
In general this method should be made more readable (out of scope here I guess).

Reply:
Both points agreed upon. I will not open an issue for it though since it will be revisited in any case during the Eve integration